### PR TITLE
Tightened role sorting function. Remove redundant unit test.

### PR DESCRIPTION
**Pull request by AMcCartneyMTH**
Created and applied a sort function for role association.

### CODE REVIEW

**Review by stottidfe — APPROVED** [review]
good job

File: src/app/users/associateRoles.js [hunk 1]
```diff
@@ -26,7 +25,7 @@ const getViewModel = async (req) => {
   const userOrganisations = userId.startsWith('inv-') ? await getInvitationOrganisations(userId.substr(4), req.id) : await getUserOrganisations(userId, req.id);
   const organisationDetails = userOrganisations.find(x => x.organisation.id === req.params.orgId);
   const policyEngineResult = await policyEngine.getPolicyApplicationResultsForUser(userId.startsWith('inv-') ? undefined : userId, req.params.orgId, req.params.sid, req.id);
-  const serviceRoles = policyEngineResult.rolesAvailableToUser;
+  const serviceRoles = await policyEngineResult.rolesAvailableToUser;
```
Comment (kruncher) [inline src/app/users/associateRoles.js]: Is `await` necessary here? Looking at the source code in policy engine this seems to be a vanilla array.

**Comment by kruncher** [comment]
It might be nice to rebase this branch so that the commits are essentially squashed into one change.

File: src/app/users/associateRoles.js [hunk 2]
```diff
@@ -38,7 +37,7 @@ const getViewModel = async (req) => {
     organisationDetails,
     selectedRoles,
     serviceDetails,
-    serviceRoles,
+    serviceRoles: serviceRoles.sort((a, b) => a.name.localeCompare(b.name)),
```
Comment (kruncher) [inline src/app/users/associateRoles.js:40]: Should there be a unit test to capture this new sorting behaviour?

**Pull request merged**